source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRContentStackView.mm

```diff
@@ -56,25 +59,47 @@ - (ContainerStyle)getStyle
     return _style;
 }
 
-- (void)setBackgroundColor:(ContainerStyle)style
-                hostConfig:(std::shared_ptr<HostConfig> const &)config
++ (UIColor *)colorFromString:(const std::string&)colorString
 {
-    _style = style;
-    long num = 0;
-    if(style == ContainerStyle::Emphasis)
-    {
-        num = std::stoul(config->containerStyles.emphasisPalette.backgroundColor.substr(1), nullptr, 16);
-    }
-    else
-    {
-        num = std::stoul(config->containerStyles.defaultPalette.backgroundColor.substr(1), nullptr, 16);
-    }
+    long num = std::stoul(colorString.substr(1), nullptr, 16);
+
+    return [UIColor colorWithRed:((num & 0x00FF0000) >> 16) / 255.0
+                    green:((num & 0x0000FF00) >>  8) / 255.0
+                     blue:((num & 0x000000FF)) / 255.0
+                    alpha:((num & 0xFF000000) >> 24) / 255.0];
+}
+
++ (CGFloat)borderWidthFromString:(const std::string&)thicknessString
+{
+    return std::stof(thicknessString);
+}
+
+- (ContainerStyleDefinition&)paletteForHostConfig:(std::shared_ptr<HostConfig> const &)config
+{
+    return (_style == ContainerStyle::Emphasis)
+        ? config->containerStyles.emphasisPalette
+        : config->containerStyles.defaultPalette;
+}
+
+- (void)setBackgroundColorWithHostConfig:(std::shared_ptr<HostConfig> const &)config
+{
+    UIColor *color = [[self class] colorFromString:[self paletteForHostConfig:config].backgroundColor];
+
+    self.backgroundColor = color;
+}
+
+- (void)setBorderColorWithHostConfig:(std::shared_ptr<HostConfig> const &)config
+{
+    UIColor *color = [[self class] colorFromString:[self paletteForHostConfig:config].borderColor];
+
+    [[self layer] setBorderColor:[color CGColor]];
+}
+
+- (void)setBorderThicknessWithHostConfig:(std::shared_ptr<HostConfig> const &)config
+{
+    const CGFloat borderWidth = [[self class] borderWidthFromString:[self paletteForHostConfig:config].borderThickness];
```

Comment:
nit: do we really need to make it as method call?
borderWidthFromString seems to be simple and used only once  so it seems to make it inline.

Reply:
Yes, I believe this makes our code more readable. We have all these functions in method calls. If we want to end up with static methods, then we would need to pass _style to all of these functions. Please note that we need to get the palette corresponding to the given style.
- setBackgroundColorWithHostConfig
- setBorderColorWithHostConfig
- setBorderThicknessWithHostConfig

Reply:
I agree. I meant for borderWidthFromString method.

---
In reply to: [169716955](https://github.com/Microsoft/AdaptiveCards/pull/1150#discussion_r169716955) [](ancestors = 169716955)

Reply:
I think I just followed the same pattern as in setBackgroundColorWithHostConfig. I'm fine to remove the helper function if it harms the readability. We shouldn't care too much about perf - this call isn't called too frequently. Readability is more important as far as perf is no concern.